wp-cli/class-theme-check-cli.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+if ( defined( 'WP_CLI' ) && WP_CLI ) {
+	$parent_dir = dirname( __DIR__ );
+	require_once $parent_dir . '/checkbase.php';
+	require_once $parent_dir . '/main.php';
+	WP_CLI::add_command( 'theme-check', 'Theme_Check_CLI' );
+}
+
+class Theme_Check_CLI {
+	/**
+	 * Run a theme check on the specified theme or the current theme.
+	 *
+	 * @param array $args       Indexed array of positional arguments.
+	 * @param array $assoc_args Associative array of options.
+	 * @return void
+	 *
+	 * ## OPTIONS
+	 * [<theme>]
+	 * : The slug of the theme to check. If not provided, checks the current theme.
+	 *
+	 * [--format=<format>]
+	 * : Output format. Accepts 'cli' or 'json'. Default: 'cli'.
+	 *
+	 * ## EXAMPLES
+	 *     wp theme-check run
+	 *     wp theme-check run twentytwentyfour
+	 *     wp theme-check run --format=json
+	 *     wp theme-check run twentytwentyfour --format=json
+	 */
+	public function run( $args, $assoc_args ) {
+		// Get the output format
+		$format = \WP_CLI\Utils\get_flag_value( $assoc_args, 'format', 'cli' );
+		if ( ! in_array( $format, array( 'cli', 'json' ) ) ) {
+			WP_CLI::error( "Invalid format. Accepts 'cli' or 'json'." );
+			return;
+		}
+
+		// Get the current theme
+		$current_theme      = wp_get_theme();
+		$current_theme_slug = $current_theme->get_stylesheet();
+
+		// Use the provided theme slug if available, otherwise use the current theme
+		$check_theme_slug = ! empty( $args[0] ) ? $args[0] : $current_theme_slug;
+
+		// Get the theme
+		$theme = wp_get_theme( $check_theme_slug );
+
+		if ( ! $theme->exists() ) {
+			if ( $format === 'json' ) {
+				$json_output = array(
+					'check-completed' => false,
+					'result'          => "Error: Theme '{$check_theme_slug}' not found.",
+					'messages'        => array(),
+				);
+				WP_CLI::log( wp_json_encode( $json_output, JSON_PRETTY_PRINT ) );
+				return;
+			}
+			WP_CLI::error( "Theme '{$check_theme_slug}' not found." );
+			return;
+		}
+
+		// Run the checks.
+		$success = run_themechecks_against_theme( $theme, $check_theme_slug );
+
+		if ( $format === 'json' ) {
+			if ( ! $success ) {
+				$json_output = array(
+					'check-completed' => false,
+					'result'          => "Error: Theme check failed for {$check_theme_slug}.",
+					'messages'        => array(),
+				);
+				WP_CLI::log( wp_json_encode( $json_output, JSON_PRETTY_PRINT ) );
```

Comment:
Note that output of `log()` is discarded to STDOUT when `--quiet` flag is passed. But here I think we should display JSON output whatever case if quiet flag is present or not.

Reply:
Ok, I used `line()` to log the results. 

I found 2 issues regarding wp-cli loggers while working on this:
- https://github.com/wp-cli/wp-cli/pull/5979
- https://github.com/wp-cli/wp-cli/issues/5980

Reply:
> But here I think we should display JSON output whatever case if quiet flag is present or not.

Not sure I agree with bypassing `--quiet`, I would expect no output at all. An alternative is to make sure that the program always exits with a descriptive code; ie. 0 for OK, non-0 for errors.

edit: On reading @matiasbenedetto's comment below, seems like the checks are not meant to return a boolean.

> It's a little bit counterintuitive, but the theme-check plugin doesn't work with the objective of outputting a boolean ("theme OK/ theme with errors"). The plugin is not designed like that. It just outputs a list of messages regarding the theme, but it never says whether this theme is OK or not. Maybe in the future, we could implement that, but that's not the case yet.

Still, I don't think bypassing `--quiet` is cool.

Reply:
I think this was fixed on the latest commits.